corehq/util/couch_helpers.py

```diff
@@ -170,6 +172,20 @@ def _get_key_kwargs(self, key):
             }
 
 
+class TimestampDocument(Document):
+
+    last_modified = DateTimeProperty()
+    created_on = DateTimeProperty(default=datetime(year=1900, month=1, day=1))
+
+    def save(self, *args, **kwargs):
+        utcnow = datetime.utcnow()
+        self.last_modified = utcnow
+        if not self._id:
```

Comment:
this isn't foolproof. if you init a doc with an _id this'll fail. another option is to remove the default value on `created_on` and make the condition `if not self.created_on`. however that will then create false datums (i.e. the created_on would be set on save for some old models and that value would be incorrect)